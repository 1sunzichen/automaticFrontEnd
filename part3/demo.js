export const generateConfig=()=>{
    return {
        server:"http://localhost",
        port:8080,
        domain:'localhost',
        time:"20191"
    }
}

export const generateAnotherConfig=()=>{
    return{
        server:'http://localhostss',
        port:8080,
        domain:'localhost2xx',
        time:new Date()

    }
}