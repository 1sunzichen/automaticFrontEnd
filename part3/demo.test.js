import { generateConfig, generateAnotherConfig } from "./demo";
test("测试 generateConfig函数", () => {
  // expect(generateConfig()).toEqual({
  //     server:"http://localhost",
  //     port:8080,
  //     domain:'localhost'
  // })
  //快照 这功能nb
  expect(generateConfig()).toMatchSnapshot();
});
// test("测试 generateAnotherConfig函数",()=>{

//     //快照 这功能nb
//     expect(generateAnotherConfig()).toMatchSnapshot({
//         time:expect.any(Date)   //变化的数据
//     })
// })
test("测试 generateAnotherConfig函数", () => {
  //快照 这功能nb  prettier 
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date),
    },
    `
    Object {
      "domain": "localhost2xx",
      "port": 8080,
      "server": "http://localhostss",
      "time": Any<Date>,
    }
  `
  );
});
