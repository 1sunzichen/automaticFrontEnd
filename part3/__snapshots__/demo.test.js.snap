// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 generateAnotherConfig函数 1`] = `
Object {
  "domain": "localhost2xx",
  "port": 8080,
  "server": "http://localhostss",
  "time": Any<Date>,
}
`;

exports[`测试 generateConfig函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": "20191",
}
`;
